pkg/devfile/adapters/kubernetes/utils/utils_test.go

```diff
@@ -344,9 +352,76 @@ func TestUpdateContainersWithSupervisord(t *testing.T) {
 				},
 			},
 			componentType:           common.ContainerComponentType,
+			expectRunCommand:        command,
 			isSupervisordEntrypoint: true,
 			wantErr:                 true,
 		},
+		{
+			name:       "Case: custom command with single environment variable",
+			runCommand: "customRunCommand",
+			containers: []corev1.Container{
+				{
+					Name:            component,
+					Image:           image,
+					ImagePullPolicy: corev1.PullAlways,
+					Env:             []corev1.EnvVar{},
+				},
+			},
+			execCommands: []versionsCommon.Exec{
+				{
+					Id:          "customRunCommand",
+					CommandLine: command,
+					Component:   component,
+					WorkingDir:  workDir,
+					Group:       &execRunGroup,
+					Env: []versionsCommon.Env{
+						versionsCommon.Env{
+							Name:  "env1",
+							Value: "value1",
+						},
+					},
+				},
+			},
+			componentType:           common.ContainerComponentType,
+			expectRunCommand:        "env1=\"value1\" && " + command,
+			isSupervisordEntrypoint: true,
+			wantErr:                 false,
+		},
+		{
+			name:       "Case: custom command with multiple environment variable",
```

Comment:
please add cases for the debug commands as well

Reply:
updated